controllers/dockercluster_controller.go

```diff
@@ -38,88 +46,131 @@ type DockerClusterReconciler struct {
 // +kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=dockerclusters,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=dockerclusters/status,verbs=get;update;patch
 
-// Reconcile handles DockerCluster events
-func (r *DockerClusterReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, rerr error) {
+// Reconcile reads that state of the cluster for a DockerCluster object and makes changes based on the state read
+// and what is in the DockerCluster.Spec
+func (r *DockerClusterReconciler) Reconcile(req reconcile.Request) (_ reconcile.Result, reterr error) {
 	ctx := context.Background()
-	log := r.Log.WithValues("dockercluster", req.NamespacedName)
-	log.Info("Reconciling cluster")
+	log := log.Log.WithName(controllerName).
+		WithValues("namespace", req.Namespace, "dockerCluster", req.Name)
 
-	dockerCluster := &infrastructurev1alpha2.DockerCluster{}
+	// Fetch the DockerCluster instance
+	dockerCluster := &infrav1.DockerCluster{}
 	if err := r.Client.Get(ctx, req.NamespacedName, dockerCluster); err != nil {
 		if apierrors.IsNotFound(err) {
 			return ctrl.Result{}, nil
 		}
-		log.Error(err, "failed to get DockerCluster")
 		return ctrl.Result{}, err
 	}
 
+	log = log.WithValues("version", dockerCluster.APIVersion)
```

Comment:
Do we actually need to log the API version here?

Reply:
not sure, @fabriziopandini did you find this useful when testing?

Reply:
copy paste from CAPA, removing